src/game/network/gameinfo.cpp

```diff
@@ -13,7 +13,869 @@
+
+    for (int i = 0; i < MAX_SLOTS; i++) {
+        if (i != Get_Local_Slot_Num()) {
+            if (Get_Const_Slot(i)->Is_Human()) {
+                return false;
+            }
+
+            if (Get_Const_Slot(i)->Is_AI()) {
+                if (Is_Slot_Local_Ally(Get_Const_Slot(i))) {
+                    return false;
+                }
+
+                ret = true;
+            }
+        }
+    }
+
+    return ret;
+}
+
+bool GameInfo::Is_Multi_Player()
+{
+    for (int i = 0; i < MAX_SLOTS; i++) {
+        if (i != Get_Local_Slot_Num()) {
+            if (Get_Const_Slot(i)->Is_Human()) {
+                return true;
+            }
+        }
+    }
+
+    return false;
+}
+
+bool GameInfo::Is_Sandbox()
+{
+    int index = Get_Local_Slot_Num();
+    int team = Get_Const_Slot(index)->Get_Team_Number();
+
+    for (int i = 0; i < MAX_SLOTS; i++) {
+        if (i != index) {
+            const GameSlot *slot = Get_Const_Slot(i);
+
+            if (slot->Is_Occupied() && (slot->Get_Team_Number() < 0 || slot->Get_Team_Number() != team)) {
+                return false;
+            }
+        }
+    }
+
+    return true;
+}
+
+SkirmishGameInfo::SkirmishGameInfo()
+{
+    for (int i = 0; i < MAX_SLOTS; i++) {
+        Set_Slot_Pointer(i, &m_gameSlot[i]);
+    }
+}
+
+void SkirmishGameInfo::Xfer_Snapshot(Xfer *xfer)
+{
+    unsigned char version = 4;
+    xfer->xferVersion(&version, 4);
+    xfer->xferInt(&m_isPlayerPreorder);
+    xfer->xferInt(&m_crcInterval);
+    xfer->xferBool(&m_inGame);
+    xfer->xferBool(&m_inProgress);
+    xfer->xferBool(&m_unk);
+    xfer->xferInt(&m_gameID);
+    int slot_count = MAX_SLOTS;
+    xfer->xferInt(&slot_count);
+    captainslog_dbgassert(slot_count == MAX_SLOTS, "MAX_SLOTS changed, need to change version.");
+
+    for (int i = 0; i < MAX_SLOTS; i++) {
+        SlotState state = m_slot[i]->Get_State();
+        xfer->xferInt(reinterpret_cast<int *>(&state));
+        Utf16String name = m_slot[i]->Get_Name();
+
+        if (version >= 2) {
+            xfer->xferUnicodeString(&name);
+        }
+
+        bool accepted = m_slot[i]->Is_Accepted();
+        xfer->xferBool(&accepted);
+        bool unk = m_slot[i]->Get_Unk();
+        xfer->xferBool(&unk);
+        m_slot[i]->Set_Unk(unk);
+        int color = m_slot[i]->Get_Color();
+        xfer->xferInt(&color);
+        int start_pos = m_slot[i]->Get_Start_Pos();
+        xfer->xferInt(&start_pos);
+        int player_template = m_slot[i]->Get_Player_Template();
+        xfer->xferInt(&player_template);
+        int team = m_slot[i]->Get_Team_Number();
+        xfer->xferInt(&team);
+        int original_color = m_slot[i]->Get_Original_Color();
+        xfer->xferInt(&original_color);
+        int original_start_pos = m_slot[i]->Get_Original_Start_Pos();
+        xfer->xferInt(&original_start_pos);
+        int original_player_template = m_slot[i]->Get_Original_Player_Template();
+        xfer->xferInt(&original_player_template);
```

Comment:
m_IP, m_port, m_NATBehavior, m_unk2, m_unk3 not serialized? Perhaps not because it is for Skirmish Room.

Reply:
Nope.